### PR TITLE
[Core] Integration Info

### PR DESCRIPTION
**Pull request by tteschemacher**
**Description**
This PR adds the integration info class, to be able to control the integration from external parameters.

Currently, this affects only the CreateIntegrationPoints interface.

### CODE REVIEW

**Comment by tteschemacher** [comment]
At this position I want to point the question whether it is preferred to have vectors per direction within the integration_info or if it is better to have a vector of integration_info. The performance difference for the here apparent cases might be negligible, however, eventual different use cases might be affected by it.

**Comment by roigcarlo** [comment]
Hi @tteschemacher, after taking a look with @rubenzorrilla  we believe that , there should be no problem on merging this. Our only comment would be about the Integration Info class structure.

Do you really need to make it derive from flags rather than storing it as a class member? 
I understand you do this so the memory footprint is reduced, but so far only one variable is stored. I am not expert in the field so I don't know if you expect more flags being added to the Integration Info, but in case the number is low I would say it makes more sense to just store them as variable members.

**Comment by tteschemacher** [comment]
Hey @roigcarlo , many thanks for checking it.

Well, to be honest I am not sure either if it is essential. I think there will be some flags, but not very many. If it becomes too heavy we can do this split also later.

If you prefer to have this changed and having it within one class, I can definitely do this.

**Review by roigcarlo — DISMISSED** [review]
:+1: From my side

File: kratos/integration/integration_info.h [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+{
+
+///@name Kratos Classes
+///@{
+
+/// Integration information for the creation of integration points.
+/* Within this class distinct information of integration can be
+ * stored and processed.
+ */
+class IntegrationInfo : public Flags
```
Comment (roigcarlo) [inline kratos/integration/integration_info.h]:
```suggestion
class KRATOS_API(KRATOS_CORE) IntegrationInfo : public Flags
```

This should fix win errors.

Comment (tteschemacher) [reply kratos/integration/integration_info.h]: devil windows. thanks a lot for the fix!

**Comment by tteschemacher** [comment]
I runs through, now.

Thanks a lot for your review, @roigcarlo and @rubenzorrilla. And in the earlier stage in the design and development @pooyan-dadvand .

**Review by roigcarlo — APPROVED** [review]

**Comment by tteschemacher** [comment]
🥳

**Pull request merged**